rollout/canary.go

```diff
@@ -119,14 +119,17 @@ func (c *RolloutController) reconcileCanaryPause(roCtx *canaryContext) bool {
 	if currentStep.Pause == nil {
 		return false
 	}
-
-	if currentStep.Pause.Duration == nil {
+	cond := roCtx.PauseContext().GetPauseCondition(v1alpha1.CanaryPauseStep)
+	if cond == nil {
+		if !rollout.Status.ControllerPause {
+			roCtx.PauseContext().AddPauseCondition(v1alpha1.CanaryPauseStep)
```

Comment:
Please add a comment about why we conditionally add pause and how this situation would occur.

Reply:
👍